### PR TITLE
PR for lofreq/indelqual module

### PR DESCRIPTION
**Pull request by kaurravneet4123**
- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `<SOFTWARE>.version.txt` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by kaurravneet4123** [comment]
Hi @drpatelh, I hope all is well.  I have made a PR for this new module: LOFREQ/INDELQUAL. It is passing all the tests, except a linting test. The error I am getting is that 'new version of functions.nf is available'. After forking and cloning the modules repo, I typed the following command in the modules dir: 'pip install --upgrade --force-reinstall git+https://github.com/nf-core/tools.git@dev' to download the latest module template. I am just wondering if this is the correct way to get the latest module template. Kindly if you can guide me. Many thanks, Rav

**Comment by drpatelh** [comment]
Hi @kaurravneet4123 . The linting is failing because you need to update the `functions.nf` to the latest available on [`dev`](https://github.com/nf-core/tools/blob/dev/nf_core/module-template/software/functions.nf)

File: software/lofreq/indelqual/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        container "quay.io/biocontainers/lofreq:2.1.5--py38h588ecb2_4"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    file fasta
+
+    output:
+    tuple val(meta), path("*_indelqual.bam"), emit: bam
+    path "*.version.txt"                    , emit: version
```
Comment (drpatelh) [inline software/lofreq/indelqual/main.nf:26]:
```suggestion
    tuple val(meta), path("*.bam"), emit: bam
    path "*.version.txt"          , emit: version
```

File: software/lofreq/indelqual/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    lofreq indelqual \\
+        --dindel \\
```
Comment (drpatelh) [inline software/lofreq/indelqual/main.nf:33]: Is `--dindel` a mandatory argument? If not, it should be provided via `options.args`

File: software/lofreq/indelqual/main.nf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        -f $fasta \\
+        -o ${prefix}_indelqual.bam \\
```
Comment (drpatelh) [inline software/lofreq/indelqual/main.nf:35]:
```suggestion
        -o ${prefix}.bam \\
```

File: software/lofreq/indelqual/meta.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  - bam:
+      type: file
+      description: BAM file with indel qualities inserted into it
+      pattern: "*_indelqual.{bam}"
```
Comment (drpatelh) [inline software/lofreq/indelqual/meta.yml:41]:
```suggestion
      pattern: "*.{bam}"
```

File: tests/software/lofreq/indelqual/test.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+- name: lofreq indelqual
+  command: nextflow run ./tests/software/lofreq/indelqual -entry test_lofreq_indelqual -c tests/config/nextflow.config
+  tags:
+    - lofreq
+    - lofreq/indelqual
+  files:
+    - path: output/lofreq/test_indelqual.bam
```
Comment (drpatelh) [inline tests/software/lofreq/indelqual/test.yml:7]:
```suggestion
    - path: output/lofreq/test.indelqual.bam
```

File: tests/software/lofreq/indelqual/main.nf [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { LOFREQ_INDELQUAL } from '../../../../software/lofreq/indelqual/main.nf' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/software/lofreq/indelqual/main.nf:5]:
```suggestion
include { LOFREQ_INDELQUAL } from '../../../../software/lofreq/indelqual/main.nf' addParams( options: ['suffix':'.indelqual'] )
```

**Comment by kaurravneet4123** [comment]
Thanks for getting back to me @hpatel.
I will update it and push the changes.
Kind regards,
Rav


signature   Ravneet 

    On Thursday, July 1, 2021, 10:59:22 AM GMT+1, Harshil Patel ***@***.***> wrote:  
 
 


Hi @kaurravneet4123 . The linting is failing because you need to update the functions.nf to the latest available on dev

File: software/lofreq/indelqual/main.nf (hunk 1 above)
Comment (drpatelh) [reply software/lofreq/indelqual/main.nf:26]: We try not to customise the output files as much as possible to leave it to developers to do this.

**Review by drpatelh — CHANGES_REQUESTED** [review]
Looks great @kaurravneet4123 . Some minorish updates required.

**Review by drpatelh — APPROVED** [review]
Perfect 👌🏽

**Comment by kaurravneet4123** [comment]
Thanks @drpatelh :-) . Please can I check one thing: if I am working on 2 modules at the same time and want to make PRs for both of them, how should I go about it? Can I make 2 PRs from the same forked repo (or do I need to fork the modules repo twice and make the PR separately)? Your guidance would be much appreciated!

**Comment by drpatelh** [comment]
We some docs for using `git` in this way in the main README under [nf-core modules create](https://github.com/nf-core/modules#nf-core-modules-create). But basically you need to start working off branches which means you only fork the repo once and do your work on separate named branches per module - that way the changes are independent of each other.

**Comment by kaurravneet4123** [comment]
Ok. Got it. Many thanks @drpatelh :-)

**Pull request merged**